pkg/config/kpconfig.go

```diff
@@ -42,18 +44,51 @@ func (c KpConfig) DefaultRepository() (string, error) {
 
 func (c KpConfig) ServiceAccount() corev1.ObjectReference {
 	if c.serviceAccount.Name == "" {
-		return corev1.ObjectReference{Name: "default", Namespace: kpNamespace}
+		return corev1.ObjectReference{Name: "default", Namespace: kpConfigNamespace}
 	}
 
 	return c.serviceAccount
 }
 
 type KpConfigProvider struct {
-	cs k8s.ClientSet
+	client                             k8s.Interface
+	repoKey, saNameKey, saNamespaceKey string
 }
 
-func NewKpConfigProvider(cs k8s.ClientSet) KpConfigProvider {
-	return KpConfigProvider{cs: cs}
+func NewKpConfigProvider(ctx context.Context, client k8s.Interface) (KpConfigProvider, error) {
+	kpConfigMap, err := client.CoreV1().ConfigMaps(kpConfigNamespace).Get(ctx, kpConfigMapName, metav1.GetOptions{})
+	if k8serrors.IsNotFound(err) {
+		return KpConfigProvider{
+			client:         client,
+			repoKey:        defaultRepositoryKey,
+			saNameKey:      defaultServiceAccountNameKey,
+			saNamespaceKey: defaultServiceAccountNamespaceKey,
+		}, nil
+	} else if err != nil {
+		return KpConfigProvider{}, err
+	}
+
+	repoKey := defaultRepositoryKey
+	if _, ok := kpConfigMap.Data[canonicalRepositoryKey]; ok {
```

Comment:
I think this most likely most happen but, doesn't this provide precedence to canonicalRepositoryKey if both defaultRepositoryKey and canonicalRepositoryKey are set? Wouldn't we want the opposite?